src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/hearing/AdjournHearingAction.java

```diff
@@ -0,0 +1,183 @@
+    protected List<Session> sessions;
+
+    protected HearingRepository hearingRepository;
+    protected HearingPartRepository hearingPartRepository;
+    protected StatusConfigService statusConfigService;
+    protected StatusServiceManager statusServiceManager;
+    protected EntityManager entityManager;
+
+    // id & hearing part string
+    private Map<UUID, String> originalHearingParts;
+    private String previousHearing;
+
+    public AdjournHearingAction(
+        AdjournHearingRequest adjournHearingRequest,
+        HearingRepository hearingRepository,
+        HearingPartRepository hearingPartRepository,
+        StatusConfigService statusConfigService,
+        StatusServiceManager statusServiceManager,
+        ObjectMapper objectMapper,
+        EntityManager entityManager
+    ) {
+        this.adjournHearingRequest = adjournHearingRequest;
+        this.hearingRepository = hearingRepository;
+        this.hearingPartRepository = hearingPartRepository;
+        this.statusConfigService = statusConfigService;
+        this.statusServiceManager = statusServiceManager;
+        this.objectMapper = objectMapper;
+        this.entityManager = entityManager;
+    }
+
+    @Override
+    public void getAndValidateEntities() {
+        hearing = hearingRepository.findOne(adjournHearingRequest.getHearingId());
+        hearingParts = hearing.getHearingParts();
+        sessions = hearingParts.stream()
+            .map(HearingPart::getSession)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+        // Validation moved to act() due to conflict with optimistic lock
+
+        if (!statusServiceManager.canBeAdjourned(hearing)) {
+            throw new SnlEventsException("Hearing can not be adjourned");
+        }
+    }
+
+    @Override
+    public UUID[] getAssociatedEntitiesIds() {
+        val ids = hearing.getHearingParts().stream().map(HearingPart::getId).collect(Collectors.toList());
+        ids.addAll(hearing.getHearingParts().stream()
+            .map(HearingPart::getSessionId)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList()));
+        ids.add(adjournHearingRequest.getHearingId());
+
+        return ids.stream().toArray(UUID[]::new);
+    }
+
+    @Override
+    public void act() {
+        try {
+            previousHearing = objectMapper.writeValueAsString(hearing);
+        } catch (JsonProcessingException e) {
+            throw new SnlRuntimeException(e);
+        }
+
+        hearing.setStatus(statusConfigService.getStatusConfig(Status.Adjourned));
+        entityManager.detach(hearing);
+        hearing.setVersion(adjournHearingRequest.getHearingVersion());
+        hearingRepository.save(hearing);
+
+        originalHearingParts = mapHearingPartsToStrings(hearingParts);
+        hearingParts.stream().forEach(hp -> {
+            hp.setSession(null);
+            hp.setSessionId(null);
+            if (hp.getStart().isAfter(OffsetDateTime.now())) {
```

Comment:
What happens if I adjourn a hearing in the middle of a hearing part? This hearing part will get Vacated? Is it what is wanted?